src/main/java/entry/Deadline.java

```diff
@@ -0,0 +1,110 @@
+            throw new CommandFormatException(e.getMessage());
+        }
+    }
+
+    /** Parses the String to produce a LocalDateTime. If information is wrong, throws CommandFormatException,
+     * which will be eventually handled to remind users to use correct format. See Ui.
+     * @param sorted the sorted tags corresponding to various "time" arguments
+     * @return LocalDateTime
+     * @throws CommandFormatException user has typed the command in a wrong format
+     */
+    private LocalDateTime parseDate(String[] sorted) throws CommandFormatException{
+        try {
+            return LocalDateTime.of(Integer.parseInt(sorted[2]), //Day
+                    Integer.parseInt(sorted[1]), //Month
+                    Integer.parseInt(sorted[0]), //Year
+                    Integer.parseInt(sorted[3].substring(0, sorted[3].length() - 2)), //Hour
+                    Integer.parseInt(sorted[3].substring(sorted[3].length() - 2)));//Minute
+        } catch (NumberFormatException | DateTimeException | ArrayIndexOutOfBoundsException e) {
+            throw new CommandFormatException(e.getMessage());
+        }
+    }
+
+    /**
+     * @return String representation of this Deadline
+     * format [D][X] {description} ({dayofweek} {time}, {day} {month} {year})
+     */
+    @Override
+    public String toString() {
+        String type = "[D]";
+        String doneStatus = "[" + getStatusIcon() + "]";
+        String day = deadline.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.forLanguageTag("en"));
+        return String.format("%s%s %s (%s %s:%s, %s %s %s)", type, doneStatus, this.description, day,
+                this.getDeadlineHour(), this.getDeadlineMinute(), deadline.getDayOfMonth(), deadline.getMonth(),
+                deadline.getYear());
+    }
+
+    /**
+     * @return wrapper around LocalDateTime corresponding method, but well formatted
+     */
+    private String getDeadlineHour() {
+        return this.deadline.getHour() < 10 ? "0" + this.deadline.getHour() : String.valueOf(this.deadline.getHour());
+    }
+
+    /**
+     * @return wrapper around LocalDateTime corresponding method, but well formatted
+     */
+    private String getDeadlineMinute() {
+        return this.deadline.getMinute() < 10 ? "0" + this.deadline.getMinute() : String.valueOf(this.deadline.getMinute());
+    }
+
+    /***
+     * Format = {type}{done}{description}{deadline}
+     * @return String to be stored in the txt file for storage.
+     */
+    public String toStorage() {
+        //type
+        String res = "D";
+        //done status
+        res += "\u001E" + (isDone ? "T" : "F");
+        //description
+        res += "\u001E" + this.description;
+        //deadline
+        res += "\u001E" + String.format("%s %s %s %s%s", this.deadline.getDayOfMonth(), this.deadline.getMonthValue(),
+                this.deadline.getYear(), this.getDeadlineHour(), this.getDeadlineMinute());
+        return res;
+    }
+
+    LocalDateTime deadline;
```

Comment:
This field should be right after header?

Reply:
yup, thanks for pointing it out